### ISSUE TITLE
❓ [immich] Does not connect to postgress 'FATAL:  password authentication failed for user "postgres"'

### ISSUE THREAD

**Issue by Jayuuuu12** [alexbelgium/hassio-addons#1614]
The postgres addon is working. I can connect to it with the set password from a client. 

immich can't connect to it, the password I set in the config is the same one that works in the client but it seems to not be used.

Already tried setting the password in config/addon_config/immich/config.yaml, but it makes no difference.

Log when starting immich: 
(`<redacted>` is the actual password and jwt token I set)
```
-----------------------------------------------------------
 Add-on: Immich
 Self-hosted photo and video backup solution directly from your mobile phone
-----------------------------------------------------------
 Add-on version: 1.118.2
 You are running the latest version of this add-on.
 System: Home Assistant OS 13.1  (amd64 / generic-x86-64)
 Home Assistant Core: 2024.10.3
 Home Assistant Supervisor: 2024.10.2
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums
-----------------------------------------------------------
 Provided by: https://github.com/alexbelgium/hassio-addons 
-----------------------------------------------------------
 Defining permissions for main user : 
User UID: 1000
User GID : 1000
-----------------------------------------------------------
/etc/cont-init.d/00-global_var.sh: executing
DB_DATABASE_NAME='immich'
DB_HOSTNAME='db21ed7f-postgres'
DB_PASSWORD=******
DB_PORT='5432'
DB_USERNAME='postgres'
DISABLE_MACHINE_LEARNING='false'
JWT_SECRET='<redacted>'
PGID='1000'
PUID='1000'
TZ='Europe/Paris'
data_location='/share/immich'
/etc/cont-init.d/00-local_mounts.sh: executing
/etc/cont-init.d/00-smb_mounts.sh: executing
/etc/cont-init.d/01-config_yaml.sh: executing
Setting permissions for the config.yaml directory

Load environment variables from /config/addons_config/immich/config.yaml if existing
If accessing the file with filebrowser it should be mapped to /homeassistant/addons_config/immich/config.yaml
---------------------------------------------------------
Wiki here on how to use : github.com/alexbelgium/hassio-addons/wiki/Add‐ons-feature-:-add-env-variables

DB_PASSWORD='<redacted>'
/etc/cont-init.d/01-custom_script.sh: executing
Execute /homeassistant/addons_autoscripts/immich.sh if existing
Wiki here : github.com/alexbelgium/hassio-addons/wiki/Add-ons-feature-:-customisation
/etc/cont-init.d/20-folders.sh: executing
[00:19:38] INFO: Setting data location
... check /share/immich folder exists
... setting permissions
... correcting official script
/etc/cont-init.d/99-run.sh: executing
[00:19:38] INFO: Setting variables
DB_DATABASE_NAME='immich'
DB_HOSTNAME='db21ed7f-postgres'
DB_PASSWORD=******
DB_PORT='5432'
DB_USERNAME='postgres'
DISABLE_MACHINE_LEARNING='false'
JWT_SECRET='<redacted>'
PGID='1000'
PUID='1000'
TZ='Europe/Paris'
data_location='/share/immich'
[00:19:38] INFO: Defining database
[00:19:38] INFO: -----------------
[00:19:38] INFO: Connecting to external postgresql
[00:19:38] INFO: 
psql: error: connection to server at "db21ed7f-postgres" (172.30.33.5), port 5432 failed: FATAL:  password authentication failed for user "postgres"
 
Starting the upstream container
 
[mod-init] Running Docker Modification Logic
[mod-init] Adding imagegenius/mods:universal-redis to container
[mod-init] Downloading imagegenius/mods:universal-redis from registry-1.docker.io
[mod-init] Installing imagegenius/mods:universal-redis
[mod-init] imagegenius/mods:universal-redis applied to container
[migrations] started
[migrations] no migrations found

╔═══════════════════════════════╗
       __  _____ _____       __
      / / |_   _/ ____|     / /
     / /    | || |  __     / /
    / /     | || | |_ |   / /
   / /     _| || |__| |  / /
  /_/     |_____\_____| /_/

  Baseimage from linuxserver.io
╠═══════════════════════════════╣
  User/Group ID:
  User UID: 1000
  User GID: 1000
╚═══════════════════════════════╝
```

**Comment by yvos** [comment]
Hi,

I have the same issue trying to get TeslaMate running. I have set the POSTGRES_PASSWORD in the configuration. Also tried setting the username but it didn't matter. 

2024-10-26 13:25:30.119 CEST [299] FATAL:  password authentication failed for user "postgres"
2024-10-26 13:25:30.119 CEST [299] DETAIL:  Connection matched pg_hba.conf line 100: "host all all all scram-sha-256"

**Comment by alexbelgium** [comment]
Hi, I think that postgres's password can't be changed after running it once. You could try with the default password, `homeassistant`

**Comment by JzBrevis** [comment]
> Hi, I think that postgres's password can't be changed after running it once. You could try with the default password, `homeassistant`

At least for me, postgres is using the correct password. I confirmed this by connecting with my own client, which only works with the expected password

**Comment by JzBrevis** [comment]
I got it to work now by resetting the postgres addon and setting up it up with the default password. If I do that, the addon is able to connect. As suspected, it seems that immich does not use the password you enter in the configuration, it always uses the password "homeassistant"

**Comment by JzBrevis** [comment]
I repeated the same process, now using a different password without special characters, this is working. The previous, unusable password contained the '!' and '%' characters.

**Comment by yvos** [comment]
Yes! Resetting the add-on and using the default pw worked for me too.

**Comment by alexbelgium** [comment]
Thanks very much ! Perhaps it's an issue with characters escape. I'll look if I can do something

**Comment by steyrblade** [comment]
It seems I have the same issue updating to immich 1.136.0.
What is meant with "Resetting the add-on"?
Do I only need to reset the postgres password? (because this seems not to work)
And why does this happen with a running system by updating the immich addon?
I would understand this issue when i would update the postgres but not not when updating immich.

**Comment by steyrblade** [comment]
can anybody help me with this issue?

**Comment by alexbelgium** [comment]
This one is finally pinned down, and @Jayuuuu12's observation back in 2024 was the key clue: the password that failed contained `!` and `%`, and a password without special characters worked. That wasn't a coincidence.

**What's happening**

The add-on connects to your database using a web-style address:

```
postgres://username:password@hostname:5432
```

In an address like that, `%` is not an ordinary character — it starts an escape sequence, the same way `%20` means a space in a normal web link. So the PostgreSQL client quietly rewrites your password before sending it. A password like `secret%de1` gets turned into something else entirely, and the server correctly replies **`password authentication failed for user "postgres"`**.

Nothing was wrong with your database, and nothing was wrong with the password you set. It just never arrived intact. That's also why it worked from a normal database client — clients don't do that rewriting — and why resetting Postgres to the plain `homeassistant` password made it start working: that password has no special characters to mangle.

The same applies to `@`, `/`, `?` and `#` in a password.

@steyrblade — this is why your setup broke on an Immich update rather than a Postgres one: it's the Immich add-on's connection code, not Postgres.

**The fix**

PR #2980 encodes the username and password properly before putting them into that address, so they arrive exactly as you typed them. Passwords containing a single quote are handled too. The two Postgres add-ons in this repository already do this — Immich was the one place that was missed.

**Once it's merged**

You'll be able to use any password you like. If you're currently working around this with a simple password there's no need to change anything, but if you'd previously given up on a password with symbols in it, that should work again.

*This is automated analysis pending @alexbelgium's review.*
